source/TimePicker.js

```diff
@@ -419,9 +414,31 @@
 					fmtParams.locale = this.locale;
 				}
 				var merFormatter = new ilib.DateFmt(fmtParams);
-				var am = ilib.Date.newInstance({hour:10});
-				var pm = ilib.Date.newInstance({hour:14});
-				this.meridiems = [merFormatter.format(am), merFormatter.format(pm)];
+				
+				this.meridiems = merFormatter.getMeridiemsRange(fmtParams);
```

Comment:
Do we need to set this as a property of `TimePicker`, or we can keep this set within just `MeridiemPicker`, so that we only have a single system of record. We are referencing `this.meridiems` on L606 (https://github.com/enyojs/moonstone/pull/2170/files#diff-0b78987cba1669c429231298cc6c83acR606), but we can simply retrieve the value from the event originator.